### DISCUSSION TITLE
M3 Quarterly Dataset

### DISCUSSION THREAD

**Discussion by Atumbo** [alan-turing-institute/sktime#7303] [Q&A] [answered]
I am trying to load M3 Quarterly Dataset tsf to a dataframe to perform hierarchical forecasting.

` df, metadata =load_tsf_to_dataframe('m3_quarterly_dataset.tsf', replace_missing_vals_with='NaN', value_column_name='series_value', return_type='pd_multiindex_hier')`

I keep getting this KeyError 
KeyError: 'quarterly'

M3 Monthly data do not have this issue.

**Comment by fkiraly** [comment]
I believe @ericjb added these recently - FYI @ericjb

**Reply by ericjb** [reply]
I believe that Atumbo's issue is not connected in any way with my fpp3 dataset PR. 

A quick web search leads me to believe that Atumbo is looking at data from the Federal Reserve db: FRED.

https://fred.stlouisfed.org/tags/series?t=m3%3Bquarterly

Atumbo, in addition to your full Python code that exposes the problem, please indicate how you created the tsf file and/or upload that file to GitHub.

**Reply by Atumbo** [reply]
This is where I downloaded .tsf -file.
https://forecastingdata.org

**Comment by fkiraly** [comment]
Meanwhile, @Atumbo, could you describe the code with full imports to allow us to reproduce the issue?

**Reply by Atumbo** [reply]
I have this:
`from sktime.datasets import load_tsf_to_dataframe
df, metadata =load_tsf_to_dataframe('mod1_quarterly_dataset.tsf', replace_missing_vals_with='NaN', value_column_name='series_value', return_type='pd_multiindex_hier')`

![image](https://github.com/user-attachments/assets/e150503d-58f6-4e46-90c7-fcd059ac6bec)

**Reply by ericjb** [reply, marked as answer]
I reproduced the problem and found a "fix" (well, a change that seems to work): 

In the file sktime/datasets/_readers_writers/tsf.py there is a section

    44      if freq is None:
    45          freq_map = {
    46              "daily": "D",
    47              "weekly": "W",
    48              "monthly": "MS",
    49              "yearly": "YS",
    50          }

After adding the missing "quarterly": "QS" to freq_map, the code works.
